base_layer/core/src/base_node/proto/chain_metadata.rs

```diff
@@ -45,9 +45,11 @@ impl TryFrom<proto::ChainMetadata> for ChainMetadata {
         let mut acc_diff = [0; ACC_DIFFICULTY_ARRAY_LEN];
         acc_diff.copy_from_slice(&metadata.accumulated_difficulty[0..ACC_DIFFICULTY_ARRAY_LEN]);
         let accumulated_difficulty = u128::from_be_bytes(acc_diff);
-        let height_of_longest_chain = metadata
-            .height_of_longest_chain
-            .ok_or_else(|| "Height of longest chain is missing".to_string())?;
+        let height_of_longest_chain = metadata.height_of_longest_chain;
+        if height_of_longest_chain == 0 {
+            return Err("Height of longest chain is missing".to_string());
```

Comment:
this cannot be an error. 0 is a valid number, its the gen height.

Reply:
Fixed.